recipes-core/images/openvario-base-image.bb

```diff
@@ -39,7 +39,6 @@ IMAGE_INSTALL = " \
     wget \    
     autofs \
     autofs-config \
-    nano \
```

Comment:
This is controversial because it reverts #24.

Reply:
The general question is: what should the OpenVario image that everybody uses really contain? I think it should be as light as possible, and we should even get rid of bash completely (because busybox should be enough).
You could provide an extended "developer" or "power user" image with more tools, but don't bloat the basic user image with stuff 99% don't need.
If nano remains in the basic image, we could/should eliminate the "magic.mgc" dependency.

Reply:
Yeah, this is a topic for a broad discussion. I'll let @mihu-ov and @linuxianer99 correct me if I'm wrong, but my impression is that conceptually OV is not just a typical consumer device, but a tinkerer-friendly one. So it includes some quality-of-life tools that tinkerers would benefit from.

We can make them installable separately with opkg and some internet-hosted ppa (like the [one I'm hosting](https://github.com/kedder/openvario-ppa)), but to use it you need to have internet on your OV box. And to have it, currently you need to buy a wifi dongle separately, have an usb keyboard attached and configure wifi through command line.

We can have a separate images for "consumer" and "tinkerer", consumer being a "bare minimum". This is also not straightforward. We already providing a separate image for each "display size" flavor of OV device. That number is multiplied by "stable/testing" flavors. "consumer/tinkerer" would multiply that number further by 2. It would be 32 or so images to build for each release. If we go this route,  we would have to come up with something to prevent this combinatoric explosion.

Furthermore, I'm making a thing called "[openvario-shell](https://github.com/kedder/openvario-shell)", which makes OV UI a bit friendlier and makes it simpler to write additional software for the device. It also includes tools to configure the wifi, so you don't have to go to command line to do that. However, it is written in python (specifically to make it simpler to develop). And if it will ever be included in stock OV image, it will pull all the python runtime with it, which might nullify your efforts cutting the image size down :) Python is a pretty heavy dependency FWICT.

Reply:
Oh, those "display size" flavors ... I saw the horrific shell code duplication in this git repo, a 400 line shell script monster of which there are 7 copies, the delta being just one line with the screen resolution. How are you able to manage this unnecessary complexity?

Also I don't understand why different images are needed for different display sizes. Another unnecessary complexity disaster! If that cannot be auto-detected, it could be a system configuration; start with a fallback resolution that works everywhere, and allow users to switch on the first boot.

And please don't include Python in the image. It's really a huge dependency which will probably double the image size.

Why not write a C++ program similar to KoboMenu (which does the same thing on the Kobo - system configuration and XCSoar launcher), and manage it inside the XCSoar git repository? That way, it can use the XCSoar GUI toolkit, all the infrastructure, and we have a consistent user interface for everything.

Reply:
> a 400 line shell script monster of which there are 7 copies, the delta being just one line with the screen resolution

Yes, that one. Not sure how/why that happened, but my intention was to eventually replace it with openvario-shell. So I never attempted to tide it up.

> Also I don't understand why different images are needed for different display sizes

The main reason is that display configuration (all the frequencies for the LVDS display driver) need to be available on boot (i.e. `u-boot` need to know them) and they are not really autodetectable from my understanding. Those are constants that are being compiled-in into `u-boot`. There's also a separate device-tree with those constants for the linux kernel. This forces us to ship a separate u-boot package (and effectively the whole separate kernel package, even though the only difference is the device-tree file) for each display size.

> Why not write a C++ program similar to KoboMenu

My main motivation for buying OV in the first place was to be able to program it. I.e. [openvario-compman](https://github.com/kedder/openvario-compman). I chose python to make the software easily hackable on your laptop. . Python allows to do it relatively easily: you can program and debug it on your laptop and then just copy files over to your OV device and run it. Cross-compiling for OV is a completely different story (maybe that's just me not familiar with C++ tools though). "Go" would be another good choice IMO (it produces statically linked native binaries), but it'd be too much effort to switch at this point.

In any case, I'm not pushing python into OV at this point, however I think it would be a good thing to ship with the openvario: ability to script the device is a very powerful one IMO.

Reply:
You could have one image which contains many DTs and many u-boots, and write a menu to switch (and reboot).
The Go language is another bloat desaster. The typical Go program weighs more than the whole OpenVario image ...
If you want scripting: XCSoar comes with a full Lua scripting engine, which weighs 200 kB or so.
About debugging on your laptop: you can debug KoboMenu on your laptop, and the same way, you could debug a C++ OVShell on your laptop.
The KoboMenu executable weighs less than 1 MB, and that includes a copy of the whole of XCSoar's GUI toolkit (statically linked). That's much more than the current set of shell scripts, but the shell script has dependencies, too, which we could eliminate, so it would be a net win.

Reply:
For me it would be okay, to remove nano ...

Reply:
I understand the goal to shrink the image but I´d rather keep nano in the image to edit config files etc. Not many people on an airfield are comfortable with vim or something the like.

Reply:
I'll try to find a way to make nano lighter, without the magic file.

Reply:
Done and force-pushed. nano itself weighs only 230 kB, but omitting the useless-for-us magic.mgc file saves 7 MB of image size.

Reply:
Thank you!

Reply:
Thanks @MaxKellermann ! Very good solution !